src/components/SearchBox.js

```diff
@@ -254,21 +255,28 @@ class SearchBox extends Component {
       renderError,
       renderNoSuggestion,
       icon,
-      value
+      value,
+      getMicInstance,
+      renderMic,
+      showVoiceSearch
     } = this.props;
+    const theme = composeThemeObject(getTheme(themePreset), this.props.theme);
```

Comment:
This would be computed on every `render` either we can add it over context ( this ) or use it in a way it is computed only once.